internal/db/sqlite/folderdb_update.go

```diff
@@ -251,6 +251,10 @@ func (s *folderDB) DropAllFiles(device protocol.DeviceID) error {
 
 	// Recalc global for the entire folder
 
+	if err := s.dropIndexIDLocked(txp, deviceIdx); err != nil {
+		return wrap(err)
+	}
```

Comment:
optional/opionionated:  
As this function is only used here and just a single SQL call, imo it would be more readable to inline it.